set.mm

```diff
@@ -645588,6 +645803,128 @@ with respect to a linear (1-dimensional) relation.  (Contributed by BJ
       VDCVBVAJSDVBTZFVDTZULUMVDCVBVAJSDVEVLFVMVETUNVCVDVEUOUPUQURGEUS $.
   $}
 
+  ${
+    $d .0. b $.  $d .x. b $.  $d B b $.  $d I b $.  $d X b $.
+    prjspnfv01.f $e |- F = ( b e. B |->
+      if ( ( b ` 0 ) = .0. , b , ( ( I ` ( b ` 0 ) ) .x. b ) ) ) $.
+    prjspnfv01.b $e |- B = ( ( Base ` W ) \ { ( 0g ` W ) } ) $.
+    prjspnfv01.w $e |- W = ( K freeLMod ( 0 ... N ) ) $.
+    prjspnfv01.t $e |- .x. = ( .s ` W ) $.
+    prjspnfv01.0 $e |- .0. = ( 0g ` K ) $.
+    prjspnfv01.1 $e |- .1. = ( 1r ` K ) $.
+    prjspnfv01.i $e |- I = ( invr ` K ) $.
+    prjspnfv01.k $e |- ( ph -> K e. DivRing ) $.
+    prjspnfv01.n $e |- ( ph -> N e. NN0 ) $.
+    prjspnfv01.x $e |- ( ph -> X e. B ) $.
+    $( Any vector is equivalent to some vector whose value on the 0th axis is
+       either ` .0. ` or ` .1. ` : value part.  (Contributed by SN,
+       13-Aug-2023.) $)
+    prjspnfv01 $p |-
+      ( ph -> ( ( F ` X ) ` 0 ) = if ( ( X ` 0 ) = .0. , .0. , .1. ) ) $=
+      ( cc0 cfv wceq co cif cv cvv fveq1 eqeq1d id fveq2d ifbieq12d ovexd ifexd
+      oveq12d fvmptd3 fveq1d iffv a1i simpr wn wa cmulr cbs cfz eqid cdr wne wf
+      wcel c0g cdif eleqtrdi eldifad frlmbasf syl2anc 0elfz ffvelrnd drnginvrcl
+      csn cn0 syl neqne syl2an3an adantr frlmvscaval drnginvrl ifeq12da 3eqtrd
+      eqtrd ) AUCJEUDZUDUCUCJUDZKUEZJWNFUDZJCUFZUGZUDZWOWNUCWQUDZUGZWOKDUGAUCWM
+      WRALJUCLUHZUDZKUEZXBXCFUDZXBCUFZUGWRBEUIMXBJUEZXDWOXBXFJWQXGXCWNKUCXBJUJZ
+      UKXGULZXGXEWPXBJCXGXCWNFXHUMXIUQUNUBAWOJWQBUIUBAWPJCUOUPURUSWSXAUEAWOUCJW
+      QUTVAAWOWNWTKDAWOVBAWOVCZVDZWTWPWNGVEUDZUFZDXKWPIVFUDZGCXLUCHVGUFZUCGVFUD
+      ZUIJIOXNVHZXPVHZXKUCHVGUOAGVIVLZWNXPVLZXJWNKVJZWPXPVLTAXOXPUCJAXOUIVLJXNV
+      LZXOXPJVKAUCHVGUOAJXNIVMUDWBZAJBXNYCVNUBNVOVPZXNGIXOXPUIJOXRXQVQVRAHWCVLU
+      CXOVLZUAHVSWDZVTZWNKWEZXPGFWNKXRQSWAWFAYBXJYDWGAYEXJYFWGPXLVHZWHAXSXTXJYA
+      XMDUETYGYHXPGXLDFWNKXRQYIRSWIWFWLWJWK $.
+  $}
+
+  ${
+    $d B x y $.  $d X l x y $.  $d W l x y $.  $d .x. l x y $.  $d S l $.
+    $d I l x y $.  $d ph x y $.  $d .0. x y $.  $d B b $.  $d X b $.
+    $d .0. b $.  $d .x. b $.  $d I b $.  $d ph b $.
+    prjspner01.e $e |- .~ = { <. x , y >. |
+      ( ( x e. B /\ y e. B ) /\ E. l e. S x = ( l .x. y ) ) } $.
+    prjspner01.f $e |- F = ( b e. B |->
+      if ( ( b ` 0 ) = .0. , b , ( ( I ` ( b ` 0 ) ) .x. b ) ) ) $.
+    prjspner01.b $e |- B = ( ( Base ` W ) \ { ( 0g ` W ) } ) $.
+    prjspner01.w $e |- W = ( K freeLMod ( 0 ... N ) ) $.
+    prjspner01.t $e |- .x. = ( .s ` W ) $.
+    prjspner01.s $e |- S = ( Base ` K ) $.
+    prjspner01.0 $e |- .0. = ( 0g ` K ) $.
+    prjspner01.i $e |- I = ( invr ` K ) $.
+    prjspner01.k $e |- ( ph -> K e. DivRing ) $.
+    prjspner01.n $e |- ( ph -> N e. NN0 ) $.
+    prjspner01.x $e |- ( ph -> X e. B ) $.
+    $( Any vector is equivalent to some vector whose value on the 0th axis is
+       either ` .0. ` or ` .1. ` : equivalence part.  (Contributed by SN,
```

Comment:
Any vector is equivalent to a vector whose zeroth coordinate is `.0. ` or ` .1. ` (proof of the equivalence).


Cannot these two theorems be grouped ? Maybe there are some common proof steps that could be mutualized.

Reply:
This would work, since 33 steps/46 each would be shared, but since there are different hypotheses, I don't really see a good way to combine them. (Well, the only way I can think of is `ph -> ps /\ ch` which seems clunky)

Reply:
Then maybe there is a common lemma that can be extracted ? Anyway, that's only a suggestion for a later PR. Resolving.